### ISSUE TITLE
Pull Request Review Findings

### ISSUE THREAD

**Issue by dserban** [Morphl-AI/MorphL-Model-User-Search-Intent#17]
a) Here:
https://github.com/Morphl-AI/MorphL-Model-User-Search-Intent/blob/bug-fix/remove-dask/cassandra_schema/usi_csv_cassandra_schema.cql#L12

Please don't use a mix of `varchar` and `text`, just use `text` everywhere.

b) Here:
https://github.com/Morphl-AI/MorphL-Model-User-Search-Intent/blob/bug-fix/remove-dask/cassandra_schema/usi_csv_cassandra_schema.cql#L24

Two observations:
1) Please use `double` instead of `float`.
2) Cassandra collections should be declared as frozen all the way through, at all levels.
```
CREATE TABLE morphl.usi_csv_word_embeddings (
  csv_file_date date,
  keyword text,
  embeddings frozen<list<frozen<list<double>>>>,
  PRIMARY KEY ((csv_file_date, keyword))
);
```
c) At the beginning and end of this section of code:
https://github.com/Morphl-AI/MorphL-Model-User-Search-Intent/blob/bug-fix/remove-dask/prediction/batch_inference/usi_csv_batch_inference.py#L121-L142

There are calls to `.repartition(32)`. But there is no shuffling operation between the two, just the dropping of one column.
Therefore the second call to `.repartition(32)` is superfluous.

d) A more general observation without a reference to a section of code.
This pipeline seems to get the input not from some kind of analytics engine, but instead the input seems to come from humans (an uploaded CSV file, is this correct?). If so, there needs to be a guard dog module at the ingestion layer boundary, that performs several checks.

Examples of such checks would be:
1) Make sure the uploaded file is not bigger than 1MB.
2) Make sure each line is under 1000 characters (or 2000, 3000, etc. but a reasonable length).
3) Make sure each line has the expected number of columns.
4) Make sure each column is not empty and can be parsed into the expected data type.
5) Make sure columns of type text don't contain excessive amounts of punctuation.
6) Make sure none of the words in the columns of type text are longer than 25 characters.
7) Make sure the numeric columns contain text representations of a number (and no letters).

**Comment by abarbulescu** [comment]
Yes, the data comes from a CSV uploaded to a bucket and then downloaded to the instance for the duration that it is used: https://github.com/Morphl-AI/MorphL-Model-User-Search-Intent/blob/bug-fix/remove-dask/ingestion/run_ingestion.sh.

Why can't the file be bigger than 1MB?

**Comment by dserban** [comment]
I used `1MB` as just an example. Take the average / expected size of the file upload, double it, and use that number. The point is to set a limit / to disallow arbitrarily sized uploads.

**Comment by abarbulescu** [comment]
Got it. For now all the csvs are checked before being uploaded to the bucket. I agree, for a complete pipeline we need to implement this. I will add it in another issue and it will be in a future pull request.

**Comment by anghelalexandra** [comment]
Closed by #16
